packages/ifpack2/src/Ifpack2_Details_OneLevelFactory_def.hpp

```diff
@@ -224,6 +224,113 @@ OneLevelFactory<MatrixType>::create (const std::string& precType,
   return prec;
 }
 
+template<class MatrixType>
+bool
+OneLevelFactory<MatrixType>::isSupported (const std::string& precType) const
+{
+  // precTypeUpper is the upper-case version of precType.
+  std::string precTypeUpper (precType);
+  if (precTypeUpper.size () > 0) {
+    std::locale locale;
+    for (size_t k = 0; k < precTypeUpper.size (); ++k) {
```

Comment:
This conversion to upper case appears twice, so it would make sense to factor it out into a "canonicalize" function.